nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -110,6 +110,8 @@
 }
 
 if ($action eq "boot") {
+    # /run/next-system/activate is called when systemctl soft-reboot is called
+    system("@coreutils@/bin/ln", "-sfn", "$toplevel", "/run/next-system") == 0 or exit 1;
```

Comment:
Should we do this on `"switch"` too?

Reply:
Doesn't seem necessary. The point of `next-system` is to have a link to activate on soft-reboot, and a `switch` will have already activated the new system. Even on a `soft-reboot`, reactivation shouldn't be necessary.